pkg/interop/interop_test.go

```diff
@@ -85,14 +109,38 @@ func TestServeHTTP(t *testing.T) {
 			if tc.AS != nil {
 				s.RegisterAS(tc.AS)
 			}
+
+			clientRootCAsPem, err := ioutil.ReadFile("testdata/clientca.pem")
+			if err != nil {
+				t.Fatalf("Failed to read CA: %v", err)
+			}
+			clientRootCAs := x509.NewCertPool()
+			clientRootCAs.AppendCertsFromPEM(clientRootCAsPem)
+			tlsConfig := &tls.Config{
+				ClientAuth: tls.RequireAndVerifyClientCert,
+				ClientCAs:  clientRootCAs,
+			}
+			srv := httptest.NewUnstartedServer(s)
+			srv.TLS = tlsConfig
+			srv.StartTLS()
+			defer srv.Close()
+
+			clientCert, err := tls.LoadX509KeyPair("testdata/clientcert.pem", "testdata/clientkey.pem")
+			if err != nil {
+				t.Fatalf("Failed to load client certificate: %v", err)
+			}
+			client := srv.Client()
+			client.Transport.(*http.Transport).TLSClientConfig.Certificates = []tls.Certificate{clientCert}
```

Comment:
Unfortunately this magic in testing is necessary as the `httptest` server hides the server certificate that's the root CA, so no way to construct a `TLSConfig` from scratch